<?xml version="1.0"?>

<launch>
  <!--  todo:Change this to happen in source code instead-->
  <rosparam file="$(find peg_in_hole_demo)/config/peg_in_hole_params.yaml" />
  <remap from="/wrench" to="/cartesian_compliance_controller/ft_sensor_wrench"/>

  <include file="$(find ur_robot_driver)/launch/ur10e_bringup.launch" >
    <arg name="robot_ip" default="175.31.1.137" doc="SwRI Collaborative Lab's UR10e IP Address" />
    <arg name="headless_mode" default="true" doc="At SwRI Collaborative Lab, we do not use URCap, so headless_mode is required" />

    <arg name="controllers" default="joint_state_controller force_torque_sensor_controller cartesian_compliance_controller" doc="Controllers that are activated by default."/>

    <arg name="controller_config_file" default="$(find peg_in_hole_demo)/config/ur10e_controllers.yaml" doc="Config file for defining the ROS-Control controllers for assembly application" />

  </include>

  <param name="robot_description" command="$(find xacro)/xacro '$(find peg_in_hole_demo)/urdf/workcell.urdf.xacro'" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find peg_in_hole_demo)/config/basic_config_example.rviz" required="true" />

  <node type="spiral_search_node" name="spiral_search_node" pkg="peg_in_hole_demo" output="screen" required="true" />

  <!--node type="mutable_transform_pub" name="mutable_tf_pub" pkg="mutable_transform_publisher" _yaml_path:="$(find peg_in_hole_demo)/config/peg_in_hole_params.yaml"  /-->  
  <!-- rosrun mutable_transform_publisher mutable_transform_pub _yaml_path:=<PATH_TO_YOUR_YAML>-->

</launch>



